### PR TITLE
[app_dart] Add logs for postsubmit build results

### PR DESCRIPTION
**Pull request by CaseyHillers**
This is to help with debugging why the Cocoon build dashboard isn't updating.

https://flutter-dashboard.appspot.com/#/build?repo=cocoon&branch=main

### CODE REVIEW

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart [hunk 1]
```diff
@@ -53,20 +53,28 @@ class PostsubmitLuciSubscription extends SubscriptionHandler {
     final BuildPushMessage buildPushMessage =
         BuildPushMessage.fromJson(json.decode(String.fromCharCodes(base64.decode(data))) as Map<String, dynamic>);
     log.fine(buildPushMessage.userData);
+    log.fine('Updating buildId=${buildPushMessage.build?.id} for builder=${buildPushMessage.build?.result}');
```
Comment (yusuf-goog) [inline app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:56]: does .result contain the builder name?

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart [hunk 2]
```diff
@@ -53,20 +53,28 @@ class PostsubmitLuciSubscription extends SubscriptionHandler {
...
+    if (buildPushMessage.userData == null) {
+      log.fine('User data is empty');
+      return Body.empty;
+    }
     final Map<String, dynamic> userData = jsonDecode(buildPushMessage.userData!) as Map<String, dynamic>;
     final String? taskKey = userData['task_key'] as String?;
     if (taskKey == null) {
       throw const BadRequestException('userData does not contain task_key');
     }
+    log.fine('Looking up key...');
```
Comment (yusuf-goog) [inline app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:70]: Maybe print taskKey, or some value to track which key its trying to lookup?

**Review by yusuf-goog — APPROVED** [review]
LGTM, with couple comments.

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:72]: It's logged in the userData line above

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:57]: No, it's an enum of cancelled, success, infra_failure, failure

**Comment by CaseyHillers** [comment]
I did a test version with these logs and was able to debug why the postsubmit builds aren't being processed. I added the quick fix in https://github.com/flutter/cocoon/pull/1601/commits/54bcf35f6af360783dd825571f34ca73c837e820

tl;dr It was expecting encoded keys, but I decided to simplify the process by using the raw id (so we can eventually remove the KeyHelper code)

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart (hunk 1 above)
Comment (yusuf-goog) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:53]: Then change `...for builder=` to something like `...with result=`?

Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:53]: Ah, thanks! Done
